http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Http2IT.java

```diff
@@ -0,0 +1,219 @@
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.quarkus.test.services.URILike;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+
+@QuarkusScenario
+@ExtendWith(VertxExtension.class)
+public class Http2IT {
+    private static final Logger LOG = Logger.getLogger(Http2IT.class);
+    @QuarkusApplication(classes = { MorningResource.class }, properties = "http2.properties")
+    static RestService app = new RestService();
+
+    private static URILike baseUri;
+    static private Vertx vertx;
+
+    private static final String BASE_ENDPOINT = "/morning";
+    private static final String GREETING = "Buenos dias";
+
+    private static HttpClient httpClient;
+
+    @BeforeAll
+    static void setUp() {
+        baseUri = app.getURI();
+        vertx = Vertx.vertx();
+    }
+
+    @Test
+    @DisplayName("HttpClient Vertx compatibility HTTP/1.1 Test")
+    public void httpClientVertxHttp1(VertxTestContext vertxTestContext) {
+        Checkpoint requestCheckpoint = vertxTestContext.checkpoint(2);
+        assertNotNull(vertx);
+        httpClient = vertx.createHttpClient();
+        assertNotNull(httpClient);
+        String expectedHTTP1Version = "HTTP_1_1";
+
+        vertxTestContext.verify(() -> httpClient.request(HttpMethod.GET, baseUri.getPort(), baseUri.getHost(), BASE_ENDPOINT)
+                .compose(req -> req.send()
+                        .compose(httpClientResponse -> {
+                            assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                            assertEquals(expectedHTTP1Version, httpClientResponse.version().toString());
+                            requestCheckpoint.flag();
+                            return httpClientResponse.body();
+                        }))
+                .onSuccess(body -> {
+                    LOG.info("Got data " + body.toString("ISO-8859-1"));
+                    assertThat("Body response", body.toString().contains(GREETING));
+
+                    requestCheckpoint.flag();
+
+                }).onFailure(Throwable::printStackTrace));
+
+    }
+
+    @Test
+    @DisplayName("HttpClient Vertx HTTP/2 protocol Test with ssl, alpn, trustAll true and verifyHost false")
+    public void httpClientVertxHttp2(VertxTestContext vertxTestContext) {
+
+        Checkpoint requestCheckpoint = vertxTestContext.checkpoint(2);
+        assertNotNull(vertx);
+        httpClient = vertx.createHttpClient(new HttpClientOptions()
+                .setSsl(true)
+                .setUseAlpn(true)
+                .setTrustAll(true)
+                .setVerifyHost(false)
```

Comment:
Please help me to understand scope of this PR - I think to test HTTP/2 with trust all = true and verify-host = false avoids anything that could run on production. This is not real-world use case, who will use this? Why not to actually test use generate the certificate and use it? I comprehend it's going to be self-signed certificate so there will be tweaks.

Reply:
I've removed old keystore, and I've created a new one, now we can remove the HttpClient option `.setVerifyHost(false)` and use the keystore.jks. However if we can also remove trust all= true we would need a valid SSL certificate from a trusted Certificate Authority (CA) , maybe using https://quarkus.io/guides/security-customization#registering-security-providers. But I don't think it's the purpose of these tests....

Reply:
sure, let's stick with trust all